### PR TITLE
Improve storage landing page.

### PR DESCRIPTION
**Pull request by coryan**
This fixes the problems with the landing page, part of the work for #1888.  I have uploaded the generated documentation to:

https://coryan.github.io/google-cloud-cpp/latest/storage/index.html

### CODE REVIEW

File: google/cloud/storage/examples/storage_quickstart.cc [hunk 1]
```diff
@@ -33,20 +33,21 @@ int main(int argc, char* argv[]) try {
...
-              gcs::BucketMetadata()
-                  .set_location("us-east1")
-                  .set_storage_class(gcs::storage_class::Regional()))
-          .value();
+  google::cloud::StatusOr<gcs::BucketMetadata> bucket_metadata =
+      client.CreateBucketForProject(
+          bucket_name, project_id,
+          gcs::BucketMetadata()
+              .set_location("us-east1")
+              .set_storage_class(gcs::storage_class::Regional()));
```
Comment (frankyn) [inline google/cloud/storage/examples/storage_quickstart.cc]: Add a new line here.

File: google/cloud/storage/doc/storage-main.dox [hunk 2]
```diff
@@ -102,6 +130,15 @@ GCS C++ Client library should be easy if you use either.
 
 #### Integrating with Bazel
 
+Select the [google-cloud-cpp release](https://github.com/googleapis/google-cloud-cpp/releases)
+you want to use. Download the tarball from that page and make a note of the
+SHA256 checksum of this release:
+
+```{.sh}
+curl -s https://github.com/google-cloud-cpp/archive/v0.5.0.tar.gz | sha256sum
+0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5  -
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/storage/doc/storage-main.dox]: I just did for v0.5.0, will send a separate PR to update the instructions on how to create the release notes.

File: google/cloud/storage/examples/storage_quickstart.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/examples/storage_quickstart.cc]: You want a second one?  There is a blank line before the `if (!bucket_metadata)` statement.

**Comment by coryan** [comment]
PTAL.

File: google/cloud/storage/examples/storage_quickstart.cc (hunk 1 above)
Comment (frankyn) [reply google/cloud/storage/examples/storage_quickstart.cc]: Acking, I missed that. Please ignore this comment.

**Pull request merged**